components/database.py

```diff
@@ -462,29 +544,41 @@ def get_current_ay():
         so just get the AY from the first entry.
         Test case will ensure that all entries in fixed mountings have the same AY
     '''
-    sql_command = "SELECT LEFT(acadYearAndSem, 8) FROM moduleMounted LIMIT(1)"
-    DB_CURSOR.execute(sql_command)
-    return DB_CURSOR.fetchone()[0]
+    try:
+        sql_command = "SELECT LEFT(acadYearAndSem, 8) FROM moduleMounted LIMIT(1)"
+        DB_CURSOR.execute(sql_command)
+        return DB_CURSOR.fetchone()[0]
+    except psycopg2.Error:
+        CONNECTION.rollback()
+        return []
```

Comment:
try block returns a string, so the except clause should return empty string instead of empty list